src/node/internal/internal_http_outgoing.ts

```diff
@@ -128,30 +142,18 @@ class MessageBuffer {
   }
 
   uncork(): void {
-    this.#corked--;
-    this._flush();
-  }
-
-  _flush(): void {
-    if (this.#corked <= 0) {
-      for (const [index, entry] of this.#data.entries()) {
-        if (!entry.written) {
-          entry.written = true;
-          this.#onWrite?.(index, entry);
-          entry.callback?.call(undefined);
-        }
-      }
+    if (this.#corked > 0) {
+      this.#corked--;
     }
```

Comment:
I see you removed the call to `this._flush()` here. Calling `uncork` is not expected to be passive. Once called, any data that was buffered while corked needs to be released. If that's happening here, it's not obvious at all. A comment in here explaining the logic here would be helpful, and if this is not actively flushing the cached data then it should be updated so that it does.